### ISSUE TITLE
golang 1.15 generates a compile error.

### ISSUE THREAD

**Issue by rbtcollins** [crossplane/provider-azure#211]
### What happened?

Pulled provider-azure to hack on a bug, ran make, got an error about unsupported go version; I have 1.15.6

### How can we reproduce it?
Use go 1.15.6 to build use the Makefile.

### What environment did it happen in?
```
make
16:03:03 [ .. ] verify dependencies have expected content
all modules verified
16:03:05 [ OK ] go modules dependencies verified
16:03:05 [FAIL] unsupported go version. Please make install one of the following supported version: 1.7|1.8|1.9|1.10|1.11|1.12|1.13|1.14
make[2]: *** [build/makelib/golang.mk:147: go.init] Error 1
make[1]: *** [build/makelib/common.mk:302: build.all] Error 2
make: *** [build/makelib/common.mk:313: build] Error 2
robertc@robertc-linux:~/src/provider-azure$ go version
go version go1.15.6 linux/amd64
robertc@robertc-linux:~/src/provider-azure$ git describe
v0.15.0-rc.0
```

**Comment by negz** [comment]
@rbtcollins Yeah, there's currently a set set of supported versions. We're considering removing that per https://github.com/upbound/build/pull/122 but in the meantime you should be able to override it by setting the `GO_SUPPORTED_VERSIONS` env var, e.g.:

```console
GO_SUPPORTED_VERSIONS=1.15 make
```

https://github.com/upbound/build/blob/197e04132f846b63d3f7ddd196b6a7f953f1cb0a/makelib/golang.mk#L56

Hopefully this unblocks you - I'm going to close this issue. Please chime in on the above PR or raise a new issue if you run into any issues!
